src/DynamoCore/Graph/Annotations/GroupStyleItemEntry.cs

```diff
@@ -0,0 +1,65 @@
+﻿namespace Dynamo.Graph.Annotations
+{
+    /// <summary>
+    /// This class represent a GroupStyle item that will be shown in the Group ContextMenu (basically a MenuItem)
+    /// </summary>
+    internal class GroupStyleItemEntry : GroupStyleBase
```

Comment:
Hmm I always assumed new class will be based on `StyleItem` class, no?

Reply:
@QilongTang well, talking about the GroupStyles in the PreferencesView, the StyleItem class is used basically for loading, showing and saving the group styles. For this case (GroupStyles in contextmenu) I created a new set of classes due that I have to show a Separator dynamically in the middle of MenuItems then in this way we can choose a style dynamically depending if is a separator or not (my idea with creating the GroupStyleItemEntry was not to affect the current functionality for PreferencesView).
Then If the idea is to reuse the StyleItem class for the GroupStyles context menu (instead of creating the GroupStyleItemEntry class) then I will need to do several changes and modify the way the PreferencesView is loading/showing and saving the info, let me know if it sounds good to you so I can proceed doing the changes.
Thanks

Reply:
Thank you @RobertGlobant20 , As I see this class is repeating a lot of the same properties and info of the other classes we created, I feel to make the separator dynamic, the cost seems almost to high for me. In general I would avoid creating classes hosting similar info. Please consider re-use the class we ha, and if proper, I am fine we do not make the separator dynamic either.

Reply:
@reddyashish @zeusongit Let me know your thoughts on this,

Reply:
@QilongTang I've done the change to re-use the StyleItem and GroupStyleItem class, this required changes in several files and also adding some properties, for more details please check the next commit: https://github.com/DynamoDS/Dynamo/pull/12667/commits/7798a6bb8288a9d1675f4df211876ab42d6702e2